### ISSUE TITLE
Ignoring Keep-Alive settings of Global Agent

### ISSUE THREAD

**Issue by mshauneu** [bitinn/node-fetch#599]
Hello,

[AWS](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-configuring-maxsockets.html) approach for configuring is more flexible.

This [line](https://github.com/bitinn/node-fetch/blob/2d0fc689c63c67dddb21adabf4e68f8e861b1d53/src/request.js#L233) prevent to use `http(s).globalAgent.options.keepAlive`

**Comment by bitinn** [comment]
We might consider changing this default behavior in future, but for now, users have to manage their connection pool by passing us an agent explicitly.

Relevant spec: https://fetch.spec.whatwg.org/#connections

**Comment by gunzip** [comment]
I agree that overriding global agent options causes unexpected behavior for NodeJS devs so `getNodeRequestOptions` should onor them. This is expecially true where you have to use node-fetch aside other http clients (ie. some cloud sdk) that you want to configure once.

**Comment by Prinzhorn** [comment]
I think https://github.com/node-fetch/node-fetch/pull/1736 kind of fixed this? At least the fact that the underlying `keepAlive` of the default agent was previously broken by explicitly setting `Connection: close`.

Or are there any other gotchas when patching the `globalAgent`?
